### ISSUE TITLE
Grouped Product Frontend Quantity Validations are Missing

### ISSUE THREAD

**Issue by Mohamed-Asar** [magento/magento2#39479]
### Preconditions and environment

- Magento version 2.4.7-p3

### Steps to reproduce

1. Create a Grouped Product and associate it with child products.
2. Set the following quantity properties for one of the child products:

    - Minimum Qty Allowed in Shopping Cart: 2
    - Maximum Qty Allowed in Shopping Cart: 10
    - Qty Increments: 2

3.Open the Product Detail Page (PDP) of the Grouped Product and test invalid quantities for the selected child product (e.g., 1, 3, or 11) to confirm that invalid inputs are restricted.

### Expected result

The validation error should be displayed upon clicking the "Add to Cart" button, without reloading the page, similar to the behavior on the Simple Product PDP.

![image](https://github.com/user-attachments/assets/0cd849ab-ff6b-439d-9d67-36fb0392542f)

### Actual result

Instead of displaying the error immediately, the page reloads and the error is shown afterward from the server side.

![image](https://github.com/user-attachments/assets/c9eeaaab-7af3-4e01-97da-cf15ad1fcb08)

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by Mohamed-Asar** [comment]
@magento i'm working on this

**Comment by engcom-Bravo** [comment]
Hi @Mohamed-Asar,

Thanks for your reporting and collaboration.

We have verified the issue in Latest 2.4-develop instance and the issue is reproducible.Kindly refer the screenshots.

**Steps to reproduce**

- Create a Grouped Product and associate it with child products.
- Set the following quantity properties for one of the child products:

1.  Minimum Qty Allowed in Shopping Cart: 2
2.  Maximum Qty Allowed in Shopping Cart: 10
3.  Qty Increments: 2


- Open the Product Detail Page (PDP) of the Grouped Product and test invalid quantities for the selected child product (e.g., 1, 3, or 11) to confirm that invalid inputs are restricted.

<img width="1424" alt="Screenshot 2024-12-16 at 10 50 27 am" src="https://github.com/user-attachments/assets/1da31aae-0f7b-4e3a-afb8-af1eb58d8e35" />


Instead of displaying the error immediately, the page reloads and the error is shown afterward from the server side.


Hence Confirming the issue.

Thanks.

**Comment by rostilos** [comment]
@magento I am working on this

**Comment by Mohamed-Asar** [comment]
@rostilos This bug was already covered in this `[PR](https://github.com/magento/magento2/pull/39480)`

**Comment by rostilos** [comment]
@Mohamed-Asar Yes, I apologize, I didn't notice at first. I'll let it go and not make edits since I see you've already left a PR
